pallets/proofs-dealer/src/utils.rs

```diff
@@ -1032,13 +1032,13 @@ impl<T: pallet::Config> ProofsDealerInterface for Pallet<T> {
         let root = ProvidersPalletFor::<T>::get_root(*provider_id)
             .ok_or(Error::<T>::ProviderRootNotFound)?;
 
-        Ok(
+        let (_, new_root, _) =
             <T::ForestVerifier as TrieProofDeltaApplier<T::MerkleTrieHashing>>::apply_delta(
                 &root, mutations, proof,
             )
-            .map_err(|_| Error::<T>::FailedToApplyDelta)?
-            .1,
-        )
+            .map_err(|_| Error::<T>::FailedToApplyDelta)?;
+
+        Ok(new_root)
```

Comment:
This change is not really necessary isn't it?

Reply:
No but I just felt what we were returning wasn't clear enough 😬